pkg/plugins/resources/postgres/listener_test.go

```diff
@@ -0,0 +1,76 @@
+package postgres
+
+import (
+	"context"
+	postgres_config "github.com/kumahq/kuma/pkg/config/plugins/resources/postgres"
+	"github.com/kumahq/kuma/pkg/core/plugins"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	kuma_events "github.com/kumahq/kuma/pkg/events"
+	core_metrics "github.com/kumahq/kuma/pkg/metrics"
+	postgres_events "github.com/kumahq/kuma/pkg/plugins/resources/postgres/events"
+	test_postgres "github.com/kumahq/kuma/pkg/test/store/postgres"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"time"
+)
+
+var _ = Describe("Events", func() {
+	var cfg postgres_config.PostgresStoreConfig
+	var listenerErrCh chan error
+	var listenerStopCh chan struct{}
+	var eventBusStopCh chan struct{}
+
+	BeforeEach(func() {
+		listenerStopCh = make(chan struct{})
+		listenerErrCh = make(chan error)
+		eventBusStopCh = make(chan struct{})
+		c, err := c.Config(test_postgres.WithRandomDb)
+		Expect(err).ToNot(HaveOccurred())
+		cfg = *c
+		ver, err := migrateDb(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ver).To(Equal(plugins.DbVersion(1677096751)))
+	})
+
+	DescribeTable("should receive a notification from pq listener",
+		func(usePgx bool) {
+			eventsBus := kuma_events.NewEventBus()
+			listener := eventsBus.New()
+			l := postgres_events.NewListener(cfg, eventsBus, usePgx)
+			go func() {
+				listenerErrCh <- l.Start(listenerStopCh)
+			}()
+
+			metrics, err := core_metrics.NewMetrics("Standalone")
+			Expect(err).ToNot(HaveOccurred())
+			pStore, err := NewStore(metrics, cfg)
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(1*time.Second)
```

Comment:
Answer to the below, I hate it but I think it fixes the deadlock. I think that's because we need some time to establish the connection in the listener and if we don't wait here the update can happen before we start listening. We could have "Eventually" waiting on the connection to be healthy but we don't have that functionality in neither PQ/PGX :(